### PR TITLE
Enable using bare pointers for GPU kernels with static shape, and then use that support

### PR DESCRIPTION
**Pull request by krzysz00**
The changes needed to make this work upstream are at https://reviews.llvm.org/D130716 and have landed - a copy of that commit is on this branch

### CODE REVIEW

**Review by sjw36 — APPROVED** [review]
Looks good. Thanks for doing that.

**Comment by krzysz00** [comment]
Update: looks like we need to poke at how MIOpen calls us...

**Comment by krzysz00** [comment]
We're now `skip-ci` because there's no point in wasting CI resources while we're blocked on working out exactly how to change MIOpen.

**Comment by krzysz00** [comment]
@pcf000 , can you help me navigate making a MIGraphX PR that would accomodate the ABI change here?

@jerryyin , is there anything in particular we need to do over in MIOpen?

**Comment by krzysz00** [comment]
(Update, @jungpark-mlir  might be the person to tag in for MIGraphX? )

**Comment by jungpark-mlir** [comment]
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/7dcae03776938bf983ef6184a13bb9bcada678b5/src/targets/gpu/mlir.cpp#L623
This is where MIGraphX puts the memref descriptor fields into the kernel arguments.

**Comment by krzysz00** [comment]
Blocked on @zhanglx13 getting his changes to finding MLIR into MIOpen, since we don't want both those changes to be one MIOpen PR https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1673

File: mlir/include/mlir-c/Dialect/MIGraphX.h [hunk 1]
```diff
@@ -18,6 +18,8 @@
 extern "C" {
 #endif
 
+#define MLIR_MIGRAPHX_DIALECT_API_VERSION 2
+
```
Comment (jungpark-mlir) [inline mlir/include/mlir-c/Dialect/MIGraphX.h:24]: Can we have a comment here to keep the features added/deleted per each version?

**Review by jungpark-mlir — APPROVED** [review]

**Pull request merged**